core/setup.py

```diff
@@ -39,6 +39,10 @@
 extras_require = {
     'ray': [
         "ray>=1.10,<1.11",
+        # https://github.com/protocolbuffers/protobuf/issues/10051
+        # protobuf made breaking change and ray doesn't upper cap its version
+        # need this hotfix for now until ray provides a new release
+        "protobuf==3.20.1"
```

Comment:
Should we use `protobuf<=3.20.1` or `protobuf==3.20.1`?

Reply:
imo `protobuf<4` is be the safer option in case of security fixes for v3
```suggestion
        "protobuf<4"
```
at least until ray releases 1.12.2 or 1.13.0

Reply:
@ddelange good suggestion, added